src/hashmap/hashmap.ts

```diff
@@ -0,0 +1,119 @@
+type Key = number | string
+
+interface IItem<T> {
+  key: Key;
+  value: T;
+}
+
+interface IHashmap<T> {
+  search: (key: Key) => T | null;
+  insert: (key: Key, value: T) => boolean;
+  delete: (key: Key) => T | null;
+}
+
+export class Hashmap<T> implements IHashmap<T> {
```

Comment:
First time I see a realization that doesn't allow me to add 5 elements into HashMap, if it initial size is 5 ((
Overall realisation is just too simple and doesn't take into account different cases like collisions etc. Also the hash method doesn't look robust also.